solutions/usaco-839.mdx

```diff
@@ -0,0 +1,57 @@
+---
+id: usaco-839
+source: USACO Gold 2018 Open
+title: Talent Show
+author: Arpan Banerjee
+---
+
+[Official Editorial](http://www.usaco.org/current/data/sol_talent_gold_open18.html)
+
+<Warning title="Common Misconception">
+Instead of using binary search as the official solution does, a knapsack can be done. An array of talents can be kept, where `talent[i]` is the minimum weight required to create a talent sum of i. To make sure each cow is used at most once and the order of the cows doesn't matter, the indices of `talent` are iterated though from greatest to least, and the `cows` are iterated though in the outer loop; the implementation should make this clear. This maximizes the ratio $t/w$ as desired. Then, the maximum ratio of $t_i / w_i$ can be easily found by going through `talent` and ignoring any $w_i$ that is less than $W$. 
+
+My $\mathcal{O}(\text{Total talent } \cdot n)$ code is below:
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include<bits/stdc++.h>
+#define w first
+#define t second
+using namespace std;
+
+const int inf=2e9;
+int n,w; double ans=0;
+
+signed main(){
+       ifstream cin("talent.in"); ofstream cout("talent.out");
+       cin>>n>>w;
+       pair<int, int> cows[n];
+       for(int i=0; i<n; i++) cin>>cows[i].w>>cows[i].t;
+       vector<int> talent(1e6+10, inf); talent[0]=0;
+       for(auto cow:cows)
+              for(int i=250*1e3; i>=0; i--) // 250 * 10^3 is total talent
```

Comment:
Perhaps store `250 * 1e3` in a variable...

Reply:
done